### PR TITLE
Slightly nicer error messages re: zerotier-one

### PR DESCRIPTION
**Pull request by laduke**
Hiya, 
I'm just hacking and looking at the book and letting my editor guess some stuff for me. Curious about your opinions code-wise and error-message-wise. No rush or anything. 

image

----

User needs to have a running instance of zerotier-one and
and it needs to be joined to the network they want to
server dns on.

closes issue #104

some example output

``` shell
travis@tl-zt zeronsd %  sudo  ./target/debug/zeronsd start $NETWORK
INFO - Welcome to ZeroNS!
ERROR - Can't connect to zerotier-one at http://localhost:9993. Is it installed and running?

travis@tl-zt zeronsd % zt-load
travis@tl-zt zeronsd %  sudo  ./target/debug/zeronsd start $UNJOINED_NETWORK
INFO - Welcome to ZeroNS!
ERROR - Are you joined to a84ac5c10a744c9d?

travis@tl-zt zeronsd %  sudo   ./target/debug/zeronsd start -t /tmp/bad-token $JOINED_NETWORK
INFO - Welcome to ZeroNS!
ERROR - error in response: status code 403 Forbidden

travis@tl-zt zeronsd %  sudo ./target/debug/zeronsd start -t /tmp/good-token $JOINED_NETWORK
INFO - Welcome to ZeroNS!
(it works normally)
```

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -117,15 +117,24 @@ pub(crate) async fn get_listen_ips(
...
 
-    let listen =
-        zerotier_one_api::apis::network_api::get_network(&configuration, network_id).await?;
-    if let Some(assigned) = listen.assigned_addresses {
-        if assigned.len() > 0 {
-            return Ok(assigned);
+    match zerotier_one_api::apis::network_api::get_network(&configuration, network_id).await {
+        Err(error) => {
+            match error {
+                zerotier_one_api::apis::Error::ResponseError(_) => Err(anyhow!("Are you joined to {}?", network_id)),
```
Comment (erikh) [inline src/utils.rs:123]: we should probably be checking the status codes here

**Review by erikh — COMMENTED** [review]
lgtm, some food for thought if you want to noodle some more. I'll merge tomorrow.

File: src/utils.rs (hunk 1 above)
Comment (laduke) [reply src/utils.rs:123]: Yeah I agree. I'll take another stab at it. 
403 unauthorized could use it's own nice message.

**Comment by erikh** [comment]
I didn't merge this yesterday. Gonna merge now, feel free to patch it again! :D

**Pull request merged**